mne_icalabel/tests/test_features.py

```diff
@@ -0,0 +1,354 @@
+    psdmed = _eeg_rpsd_compute_psdmed(epochs, icaact, ncomp, nfreqs, n_points,
+                                      nyquist, index, window, subset_eeglab)
+
+    psdmed_eeglab = loadmat(psd_psdmed_epo_path)['psdmed']
+    assert np.allclose(psdmed, psdmed_eeglab, atol=5e-2)
+    # TODO: investigate why the tolerance had to be brought this high for this
+    # particular case.
+
+
+def test_eeg_rpsd():
+    """Test eeg_rpsd function that extract the PSD feature from the IC."""
+    # Raw --------------------------------------------------------------------
+    # Compare that both MATLAB files are identical (since rng('default') was
+    # called both time, resetting the seed).
+    psd1 = loadmat(psd_steps_raw_path)['psd']
+    psd2 = loadmat(psd_raw_path)['psd']
+    assert np.allclose(psd1, psd2, atol=1e-4)
+
+    # clean-up
+    psd_eeglab = psd2.copy()
+    del psd1
+    del psd2
+
+    # compute psd in Python
+    raw = read_raw(raw_eeglab_path, preload=True)
+    ica = read_ica_eeglab(raw_eeglab_path)
+    icaact = compute_ica_activations(raw, ica)
+
+    # retrieve subset from eeglab
+    constants_eeglab = loadmat(psd_constants_raw_path)['constants'][0, 0]
+    assert constants_eeglab['subset'].shape[0] == 1
+    subset_eeglab = constants_eeglab['subset'][0, :] - 1
+
+    # retrieve the rest from python
+    ncomp, nfreqs, n_points, nyquist, index, window, _ = \
+        _eeg_rpsd_constants(raw, ica)
+
+    # compute psdmed
+    psdmed = _eeg_rpsd_compute_psdmed(raw, icaact, ncomp, nfreqs, n_points,
+                                      nyquist, index, window, subset_eeglab)
+
+    # format and compare
+    psd = _eeg_rpsd_format(psdmed)
+    assert np.allclose(psd, psd_eeglab, atol=1e-4)
+
+    # Epochs -----------------------------------------------------------------
+    # Compare that both MATLAB files are identical (since rng('default') was
+    # called both time, resetting the seed).
+    psd1 = loadmat(psd_steps_epo_path)['psd']
+    psd2 = loadmat(psd_epo_path)['psd']
+    assert np.allclose(psd1, psd2, atol=1e-4)
+
+    # clean-up
+    psd_eeglab = psd2.copy()
+    del psd1
+    del psd2
+
+    # compute psd in Python
+    epochs = read_epochs_eeglab(epo_eeglab_path)
+    ica = read_ica_eeglab(epo_eeglab_path)
+    icaact = compute_ica_activations(epochs, ica)
+
+    # retrieve subset from eeglab
+    constants_eeglab = loadmat(psd_constants_epo_path)['constants'][0, 0]
+    assert constants_eeglab['subset'].shape[0] == 1
+    subset_eeglab = constants_eeglab['subset'][0, :] - 1
+
+    # retrieve the rest from python
+    ncomp, nfreqs, n_points, nyquist, index, window, _ = \
+        _eeg_rpsd_constants(epochs, ica)
+
+    # compute psdmed
+    psdmed = _eeg_rpsd_compute_psdmed(epochs, icaact, ncomp, nfreqs, n_points,
+                                      nyquist, index, window, subset_eeglab)
+
+    # format and compare
+    psd = _eeg_rpsd_format(psdmed)
+    assert np.allclose(psd, psd_eeglab, atol=1e-5)
```

Comment:
FYI previous comment. `psdmed` from this test function and `psdmed` from the previous test do match. After formatting, a tolerance of `1e-5` works, while this is not the case before formatting.

Reply:
solved itself, probably wrong file.